src/ApiClient.php

```diff
@@ -137,24 +131,92 @@ public function getRentalUnit(int $rentalUnitId): RentalUnit
 
         $result = $this->runQuery($query);
 
-        return $this->parseResult($result, RentalUnit::class)->getData()->getRentalUnit();
+        return $this->parseResult($result, RentalUnitCallResponseBody::class)->getData()->getRentalUnit();
     }
 
-    public function createOrReplaceAllotments(int $rentalUnitId, AllotmentCollection $allotmentCollection): AllotmentCollection
+    /**
+     * USAGE
+     * createOrReplaceAllotments (
+     *      1,
+     *      [
+     *          [
+     *              'date' => '2020-01-01',
+     *              'allotments' => 1,
+     *          ],
+     *          [
+     *              'date' => '2020-02-01',
+     *              'allotments' => 1,
+     *          ]
+     *      ];
+     * )
+     *
+     * @param int $rentalUnitId
+     * @param array $allotmentCollection
+     * @return AllotmentCollection
+     * @throws BadResponseException
+     */
+    public function createOrReplaceAllotments(int $rentalUnitId, array $allotmentCollection): AllotmentCollection
     {
+        $normalizedAllotmentCollection = [];
+        foreach ($allotmentCollection as $allotment) {
+            if ($allotment instanceof Allotment) {
+                $normalizedAllotmentCollection[] = $allotment->toArray();
+            } elseif (is_array($allotment)) {
+                $normalizedAllotmentCollection[] = $allotment;
+            }
+        }
+
         $mutation = MutationBuilder::createCreateOrUpdateAllotmentsMutation();
 
-        $variables = ['input' => array_merge(['rentalUnitId' => $rentalUnitId],  $allotmentCollection->toArray())];
+        $variables = ['input' => ['rentalUnitId' => $rentalUnitId, 'allotments' => $normalizedAllotmentCollection]];
         $result = $this->runQuery($mutation, $variables);
 
         return $this->parseResult($result, CreateOrReplaceAllotmentsCallResponseBody::class)->getData()->getCreateOrReplaceAllotments();
     }
 
-    public function createOrReplaceTripPricings(int $rentalUnitId, TripPricingCollection $tripPricingCollection): TripPricingCollection
+    /**
+     * USAGE
+     * createOrReplaceTripPricings (
+     *      1,
+     *      [
+     *          [
+     *              'date' => '2020-01-01',
+     *              'duration' => 1,
+     *              'price' => 100,
+     *          ],
+     *          [
+     *              'date' => '2020-02-01',
+     *              'duration' => 1,
+     *              'price' => 100.50,
+     *          ]
+     *      ];
+     * )
+     *
+     *
+     * @param int $rentalUnitId
+     * @param array $tripPricingCollection
+     * @return TripPricingCollection
```

Comment:
Imo zijn de @param en @return annotaties tegenwoordig echt overbodig als de functie parameter en return types heeft, en geven ze enkel maar kans op extra fouten bij evt. wijzigingen. Enige uitzondering is als de annotatie extra info geeft, bijvoorbeeld door een specifieke typing voor een array (bijv. TripPricing[]), of als er echt uitleg achter staat. Maar misschien zijn er redenen om ze wel altijd te hebben?

Reply:
- [ ] Verwijderen annotaties